integration_tests/src/main/python/hive_parquet_write_test.py

```diff
@@ -58,6 +60,20 @@
 fallback_nodes = ['ProjectExec'] if is_databricks_runtime() or is_spark_350_or_later() else []
 
 
+def read_single_bucket(table, bucket_id):
+    # Bucket Id string format: f"_$id%05d" + ".c$fileCounter%03d".
+    # fileCounter is always 0 in this test. For example '_00002.c000' is for
+    # bucket id being 2.
+    # We leverage this bucket segment in the file path to filter rows belong to a bucket.
+    bucket_segment = '_' + "{}".format(bucket_id).rjust(5, '0') + '.c000'
+    return with_cpu_session(
+        lambda spark: spark.sql("select * from {}".format(table))
+        .withColumn('file_name', f.input_file_name())
+        .filter(f.col('file_name').contains(bucket_segment))
+        .drop('file_name')  # need to drop the file_name column for comparison.
```

Comment:
One concern with this approach is that if it's wrong, we'll silently filter the dataframe to no rows, and the tests will pass because 0 rows == 0 rows.  Would be great to have a sanity check, like in addition to comparing the rows found in each bucket, also counting the number of rows found in each bucket and after all buckets have been visited, verifying the sum of bucket row counts is the same as the row count of the entire dataframe to make sure we don't miss any (or all) buckets.

Reply:
great idea, updated.